app/models/differential_expression_result.rb

```diff
@@ -87,11 +87,72 @@ def filename_for(label, comparison: nil)
     "#{basename}.tsv"
   end
 
-  # map of all observed result files, of label value => label-specific filenames
+  # map listing all result files and their constituent groups , by comparison type
   # this is important as it sidesteps the issue of study owners renaming clusters, as cluster_name is cached here
+  #
+  # @return [Hash<String => Array<String, String>, Array<String, String, String>]
   def result_files
-    files = one_vs_rest_comparisons.map { |label| filename_for(label) }
-    Hash[one_vs_rest_comparisons.zip(files)]
+    pairwise_files = []
+    is_author_de = false
+
+    # TODO (SCP-5096): Productionize this block, remove example data
+    if Rails.env.development? && annotation_name == 'General_Celltype'
+
+      is_author_de = true
+
+      self.one_vs_rest_comparisons = [
+        'B cells',
+        # 'CSN1S1 macrophages', # Simulate missing one-vs-rest comparison
+        'dendritic cells',
+        'eosinophils',
+        'fibroblasts',
+        'GPMNB macrophages',
+        'LC1',
+        'LC2',
+        'neutrophils',
+        'T cells'
+      ]
+
+      # Two important notes for pairwise comparisons:
+      #
+      #   1.  It conveys observed _combinations_ of groups.  Order does not
+      #       matter in combinations.  This means that, with a bit of extra
+      #       care, we can store 1/2 the data than a naive approach.
+      #
+      #   2.  Per (1), only the _naturally ordered_ pairs appear in the DE
+      #       results manifest file, and in this `pairwise_comparisons` data
+      #       structure.  Note that all keys in the hash below are naturally
+      #       ordered before all values for that key.
+      pairwise_comparisons = {
+        # Note that first key ('B cells') lacks comparison to 'CSN1S1 macrophages',
+        # which mimics realistic potential missing pairwise comparison in
+        # user-uploaded DE data.
+        'B cells' => ['dendritic cells', 'eosinophils', 'fibroblasts', 'GPMNB macrophages', 'LC1', 'LC2', 'neutrophils', 'T cells'],
+        'CSN1S1 macrophages' => ['dendritic cells', 'eosinophils', 'fibroblasts', 'GPMNB macrophages', 'LC1', 'LC2', 'neutrophils', 'T cells'],
+        'dendritic cells' => ['eosinophils', 'fibroblasts', 'GPMNB macrophages', 'LC1', 'LC2', 'neutrophils', 'T cells'],
+        'eosinophils' => ['fibroblasts', 'GPMNB macrophages', 'LC1', 'LC2', 'neutrophils', 'T cells'],
+        'fibroblasts' => ['GPMNB macrophages', 'LC1', 'LC2', 'neutrophils', 'T cells'],
+        'GPMNB macrophages' => ['LC1', 'LC2', 'neutrophils', 'T cells'],
+        'LC1' => ['LC2', 'neutrophils', 'T cells'],
+        'LC2' => ['neutrophils', 'T cells'],
+        'neutrophils' => ['T cells']
+      }
+
+      pairwise_comparisons.each_pair do |label, comparisons|
```

Comment:
Slightly more concise, and also matches the implementation for `one_vs_rest_files`.
```suggestion
      pairwise_files = pairwise_comparisons.map do |label, comparisons|
        comparisons.map do |comparison|
          [label, comparison, filename_for(label, comparison:)]
        end
      end
```